app/main.py

```diff
@@ -1 +1,39 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(
+            self,
+            name: str,
+            health: int = 100,
+            hidden: bool = False,
+    ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+
+        if health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden:
+            self.hidden = False
+        else:
+            self.hidden = True
```

Comment:
The logic in the `hide` method is inverted. The method should set `self.hidden` to `True` if it is currently `False`, and vice versa. Consider using `self.hidden = not self.hidden` to toggle the state.